api/src/main/java/io/opencensus/trace/propagation/B3Format.java

```diff
@@ -0,0 +1,143 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace.propagation;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.opencensus.trace.SpanContext;
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Implementation of the B3 propagation protocol. See
+ * <a href=https://github.com/openzipkin/b3-propagation>b3-propagation</a>.
+ *
+ * <p>This is only recommended to be used for backwards compatibility with systems like Zipkin.
+ * For new implementations we recommend using the {@code HttpFormat}.
+ *
+ * <p>Example of usage on the client:
+ *
+ * <pre>{@code
+ * private static final Tracer tracer = Tracing.getTracer();
+ * private static final B3Format b3Format = Tracing.getPropagationComponent().getB3Format();
+ * void onSendRequest() {
+ *   try (Scope ss = tracer.spanBuilder("Sent.MyRequest").startScopedSpan()) {
+ *     Map<HeaderName, String> b3Headers  =
+ *         b3Format.toHeaders(tracer.getCurrentContext().context());
+ *     // Send the request including the b3Headers and wait for the response.
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>Example of usage on the server:
+ *
+ * <pre>{@code
+ * private static final Tracer tracer = Tracing.getTracer();
+ * private static final B3Format b3Format = Tracing.getPropagationComponent().getB3Format();
+ * void onRequestReceived() {
+ *   // Get the B3 headers from the request.
+ *   Map<HeaderName, String> b3Headers = ...
+ *   SpanContext spanContext = b3Format.fromHeaders(b3Headers);
+ *   try (Scope ss =
+ *            tracer.spanBuilderWithRemoteParent("Recv.MyRequest", spanContext).startScopedSpan()) {
+ *     // Handle request and send response back.
+ *   }
+ * }
+ * }</pre>
+ */
+public abstract class B3Format {
+  public enum HeaderName {
+    X_B3_TRACE_ID("X─B3─TraceId"),
+    X_B3_SPAN_ID("X─B3─SpanId"),
+    X_B3_PARENT_SPAN_ID("X─B3─ParentSpanId"),
+    X_B3_SAMPLED("X─B3─Sampled"),
+    X_B3_FLAGS("X-B3-Flags");
+
+    private final String value;
+    private final String lowerValue;
+
+    private HeaderName(String value) {
+      this.value = value;
+      this.lowerValue = value.toLowerCase(Locale.US);
+    }
+
+    /**
+     * Returns the value of this header used in HTTP requests.
+     *
+     * @return the value of this header used in HTTP requests.
+     */
+    public String getValue() {
+      return value;
+    }
+
+    /**
+     * Returns the ASCII value of this header (lowercase).
+     *
+     * @return the ASCII value of this header (lowercase).
+     */
+    public String getLowerValue() {
```

Comment:
How will this method be used?

Reply:
guess is in grpc? (there's existing grpc instrumentation that use B3, and grpc downcases header names)